### ISSUE TITLE
Tags automatiques quand du code est inclus dans un message

### ISSUE THREAD

**Issue by Vayel** [zestedesavoir/zds-site#2344]
> Déjà quelque chose qui serait assez simple à mettre en place (je penses notamment aux forums Programmation et Sites web) c'est si le premier post contient du code dans tel ou tel langage on l'ajoute automatiquement comme tag, sauf si il y est déjà, bien-sûr.
> 
> Par exemple, je crées un sujet "Erreurs de compilation incompréhensible", ça peut être dans n'importe quel langage. Mais dans mon message je mets un bout de code C++ et le tag est tout suite ajouter.
> 
> De Bat', [ici](http://zestedesavoir.com/forums/sujet/2333/propositions-de-tags/?page=1#p43412)

**Comment by ghost** [comment]
Peut-être rajouter un message qui conseil de rajouter le tag plutôt que de forcer l'ajout ?

**Comment by GerardPaligot** [comment]
@cgabard Ca pourrait pas être renvoyé par zMarkdown via les méta données que tu comptes implémenté ? Si oui, il faudrait placer cette issue sur le dépôt Python-ZMarkdown.

**Comment by gustavi** [comment]
Humm, je suis pas super fan de l'issue. On peut vite se retrouver avec des sujets avec 10 tags.

**Comment by pierre-24** [comment]
@gustavi peut être alors se limiter au premier message ?

**Comment by WinXaito** [comment]
@gustavi comme le dit @pierre_24 il était préciser uniquement le premier message.

> c'est si le premier post contient du code dans tel ou tel langage on l'ajoute automatiquement comme tag

**Comment by vhf** [comment]
Il faudra prévoir une map d'alias de langages. (ping @cgabard si c'est de ton côté)

Exemple : 

``````
```py
```python
``````

sont équivalents, mais le tag doit être `python`.

**Comment by artragis** [comment]
@vhf je pense qu'une possibilité ça serait d'exporter les langages comme on exporte les pings et côté zds on joue avec ça.

**Comment by vhf** [comment]
Je peux sortir ça ouais, sans problème.

**Comment by vhf** [comment]
https://github.com/zestedesavoir/zmarkdown/pull/210

Doc ici: https://github.com/zestedesavoir/zmarkdown/blob/d00e305c4a481a30059662dc6c6790e03008b837/packages/zmarkdown/README.md#response-1

Je passe la main pour la partie backend zds-site.

**Comment by artragis** [comment]
merci. je m'en occupe asap.

**Comment by A-312** [comment]
J'aimais bien l'idée de suggérer le tag plutôt que ça soit une obligation/automatique (dans certain cas le tag du langage n'est pas souhaité). Par contre la suggestion peut se baser sur les balises codes pour être utile.

**Comment by artragis** [comment]
Non, on va utiliser les infos retournées pqr zmd pour ajouter automatiquement un tag, c'est facile à faire, logiquement.

**Comment by A-312** [comment]
C'est souhaité d'avoir un tag forcé ? Par exemple ici les tags "automatiques" ne sont pas souhaités : https://zestedesavoir.com/forums/sujet/12041/ajout-de-la-liste-des-tags-suivis-apres-la-liste-des-forums/#p198324

**Comment by artragis** [comment]
Je vois pas le rapport

**Comment by A-312** [comment]
j'ai du code html, css dans mon message et je ne veux pas pour autant qu'il ait un tag html ou css

**Comment by artragis** [comment]
ici c'est une question d'intégration. Il suffirait par exemple d'activer cette fonctionnalité sur les forums techniques.
